CONTRIBUTING.md

```diff
@@ -6,36 +6,108 @@ For major changes, please open an issue to discuss what you would like to change
 As of writing this document, the project is still in a pre-1.0 state.
 This means that the API is not yet stable and breaking changes may occur frequently.
 
-## Pull Requests
+## Prerequisites
+
+### Mandatory
+
+The following tools are mandatory.
+
+- [`rust`](https://www.rust-lang.org/tools/install) for working on the project at all.
+- [`git`](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git) for version control.
+- [`pre-commit`](https://pre-commit.com/#install) for running pre-commit hooks.
+  - You will need this to run the pre-commit hooks, which will run various checks on your code when you commit to `git`.
+  - The `pre-commit` hooks will try to automatically fix any issues they can, but some issues will require manual intervention.
+  - If your code does not pass these checks, your PR will not even be reviewed or considered for merging until you fix the issues.
+  - `pre-commit` is available via `pip` (`python -m pip install pre-commit`), `brew` (`brew install pre-commit`) on macOS, or via [`snap`](https://snapcraft.io/install/pre-commit/ubuntu) for Linux users.
+
+#### Python Specific Work
+
+If you are working on the Python bindings, you will need:
+
+- [`python`](https://www.python.org/downloads/) 3.11.
+- [`poetry`](https://python-poetry.org/docs/#installation) for managing python dependencies.
+  - `python -m pip install poetry` will install poetry in the local python environment. You could also install it globally by following the instructions in the link above.
+  - To install the dependencies for a python package, `cd` into the package directory and run `poetry install`.
+
+### Optional
+
+The following tools are optional, but recommended.
+
+- [`docker`](https://docs.docker.com/engine/install/)
+  - You will need this to test the CI/CD pipelines locally or to use Earthly to run various project commands.
+  - You may need to start the docker daemon before running any commands. You can do this with `sudo systemctl start docker`.
```

Comment:
I believe systemctl is part of systemd. It's part of most major Linux distros, but what about macOS?